### PR TITLE
Add max parallels to getcapabiltilies

### PR DESCRIPTION
**Pull request by PVince81**
If SQLite, limit parallel upload to 1

Still to be discussed @guruz (in case you change your mind)

### CODE REVIEW

File: apps/files/lib/capabilities.php [hunk 1]
```diff
@@ -11,13 +11,23 @@
...
+		} else {
+			$caps['maxparallels'] = 0;
+		}
+
 		return new \OC_OCS_Result(array(
 			'capabilities' => array(
-				'files' => array(
-					'bigfilechunking' => true,
-					),
+					'files' => $caps
```
Comment (LukasReschke) [inline apps/files/lib/capabilities.php]: Add a , at EoL to not have to edit it for future adjustments?

File: apps/files/lib/capabilities.php [hunk 2]
```diff
@@ -11,13 +11,23 @@
...
 				),
-			));
+			)
+		);
+		return new \OC_OCS_Result($caps);
```
Comment (MorrisJobke) [inline apps/files/lib/capabilities.php]: Two times a return?!? I'm confused. I guess this line can be dropped.

File: apps/files/lib/capabilities.php (hunk 1 above)
Comment (nickvergessen) [inline apps/files/lib/capabilities.php]: Also the intendation is off here

File: apps/files/lib/capabilities.php [hunk 3]
```diff
@@ -11,13 +11,23 @@
...
+
 		return new \OC_OCS_Result(array(
 			'capabilities' => array(
-				'files' => array(
-					'bigfilechunking' => true,
-					),
+					'files' => $caps
 				),
-			));
+			)
```
Comment (nickvergessen) [inline apps/files/lib/capabilities.php]: trailing `,` aswell

**Comment by DeepDiver1975** [comment]
Should we disallow syncing at all? Parallel upload can result in database locks but multiple clients from multiple users can trigger the same result.

For testing and development purpose I want to keep sqlite.

Same for low traffic usage - e.g. somebody who uses ownCloud and calendar and files platform for a pure sharing perspective. In such a case sqlite is really sufficient.

File: apps/files/lib/capabilities.php (hunk 2 above)
Comment (PVince81) [inline apps/files/lib/capabilities.php]: I shouldn't have done this in a rush...

**Comment by PVince81** [comment]
I'll only fix this PR once we agree on the approach.
I raised it in a rush for @guruz as a POC.

**Comment by karlitschek** [comment]
Hmm. In which scenario is it harmful to _try_ to do several WebDAV requests in parallel? 
Doesn´t SQLite lock properly in case a lock is needed?

**Comment by guruz** [comment]
@karlitschek the busy wait handler was removed (if i understand) when switching from mdb2 to doctrine. https://github.com/owncloud/core/pull/13615 

@PVince81 ok.. and you'll also add a flag "using_legacy_unperformant_db" or so flag in case we ever want to display a message in the client? (as discussed with @LukasReschke )

**Comment by LukasReschke** [comment]
Another "SQLite does it all wrong"-issue that breaks our mobile clients: https://github.com/owncloud/core/issues/13705

**Comment by karlitschek** [comment]
I don´t understand this at all. It is possible to process several requests at the same time by using sqlite. We only need the proper locking working. This is best practices for web application for a long time and also worked in ownCloud like that. I don´t know what is currently broken in our code so that this is not working.
A capabilities call to "prevent" parallel requests is stupid. Sorry. We need to fix our sqlite handling here.

**Comment by PVince81** [comment]
Can you guys confirm whether there was data loss involved in this situation ?
Or was this just an inconvenience for the user to see locking messages appearing too often ?

**Comment by karlitschek** [comment]
A clarification about my previous comment. With the comment "stupid" I was referring to the approach that we work around a possible server bug that can lead to data lose by telling the client to not do parallel requests. If such a bug exists in the core that parallel requests can create data lose that we have to fix this bug and not tell the clients to not do parallel requests.
This wasn´t directed to a certain person or the work of a certain person.
Sorry for using bad words.

**Comment by DeepDiver1975** [comment]
My pov on the current sqlite situation is described in here https://github.com/owncloud/core/issues/13705#issuecomment-71843849

**Comment by PVince81** [comment]
@guruz can you test with https://github.com/owncloud/core/pull/13752 with parallel upload ?
Is there a way I can test this too locally ?

**Comment by PVince81** [comment]
I tested with  https://github.com/owncloud/core/pull/13752 and can confirm that it fixes the issue with parallel uploads. Would be good if @guruz can confirm this as well. Thanks.

**Comment by PVince81** [comment]
Obsoleted by https://github.com/owncloud/core/pull/13752

**Comment by guruz** [comment]
I guess you mean obsoleted by https://github.com/owncloud/core/pull/13615 ?

**Comment by dragotin** [comment]
I agree with @karlitschek, it does not make sense to limit the maximum numbers of parallel requests to the server for whatever reason or in whatever situation. While we can prevent the client from doing parallel requests we can not prevent users to use a script (for example) to get into the same situation. So that needs to be fixed, even if sqlite is used.

That is not easy to fix as @DeepDiver1975 points out. So we were looking for a solution for the upcoming releases (server 8 and client 1.8) and thus the idea of the capabilities sounds like a compromise for now.

**Pull request closed without merging**